packages/core/upload/admin/src/components/FolderCard/FolderCard.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { pxToRem } from '@strapi/helper-plugin';
+import { Box } from '@strapi/design-system/Box';
+import { Stack } from '@strapi/design-system/Stack';
+import Folder from '@strapi/icons/Folder';
+
+import { FolderCardContext } from './FolderCardContext';
+import useId from './utils/useId';
+
+const FauxClickWrapper = styled.button`
+  height: 100%;
+  left: 0;
+  position: absolute;
+  opacity: 0;
+  top: 0;
+  width: 100%;
+
+  &:hover,
+  &:focus {
+    text-decoration: none;
+  }
+`;
+
+const StyledFolder = styled(Folder)`
+  path {
+    fill: currentColor;
+  }
+`;
+
+export const FolderCard = ({ children, id, startAction, ariaLabel, onDoubleClick, ...props }) => {
+  const generatedId = useId(id);
```

Comment:
I've copied from another card and I have to say I like it a lot because it makes it impossible to break the connection between the checkbox and the card. Since this is mainly an a11y feature it would be forgotten quickly, but the hook makes sure this is not possible.

I made the hook a little easier in 85163114d963d209be05fb924395880c6c9a1c37 though, because I think we don't need a separation between `prefix` and initial id.